File: Editor/src/main/java/com/rspsi/studio/plugin/builtin/tool/SingleSelectToolPlugin.java
package com.rspsi.studio.plugin.builtin.tool;

import com.rspsi.studio.plugin.StudioToolPlugin;
import com.rspsi.studio.theme.StudioIcons;
import com.rspsi.studio.ui.StudioPanelContext;

import java.util.EnumSet;
import java.util.Set;

/**
 * Modal tool for picking exactly one tile at a time for inspection. Lives on the floating
 * tool rail only; the Tile Inspector panel shows what it selects.
 */
public final class SingleSelectToolPlugin implements StudioToolPlugin {

    public static final String ID = "studio.tool.select.single";
    public static final String ENGINE_TOOL_ID = "selection.single";

    @Override
    public String id() {
        return ID;
    }

    @Override
    public String name() {
        return "Single Select";
    }

    @Override
    public String description() {
        return "Click a tile to select exactly one tile for inspection.";
    }

    @Override
    public String icon() {
        return StudioIcons.SELECT;
    }

    @Override
    public String toolId() {
        return ENGINE_TOOL_ID;
    }

    @Override
    public String shortcut() {
        return "S";
    }

    @Override
    public int railPriority() {
        return 10;
    }

    @Override
    public String category() {
        return "Selection";
    }

    @Override
    public Set<ToolSurface> surfaces() {
        // The docked Left Tool Rail is brush-only now (Tile Painter, Height
        // Sculptor); selection lives on the floating rail instead.
        return EnumSet.of(ToolSurface.FLOATING_TOOLBAR);
    }

    @Override
    public boolean hasContextDrawerContent() {
        return false;
    }

    @Override
    public void renderContextDrawer(StudioPanelContext context) {
        // No drawer content - selection results show in the Tile Inspector panel instead.
    }
}

File: Editor/src/main/kotlin/com/rspsi/studio/plugin/builtin/tool/SingleSelectToolPlugin.kt
package com.rspsi.studio.plugin.builtin.tool

import com.rspsi.studio.plugin.StudioToolPlugin
import com.rspsi.studio.theme.StudioIcons
import com.rspsi.studio.ui.StudioPanelContext
import java.util.EnumSet

/**
 * Native Studio projection for selecting exactly one tile for inspection.
 *
 * Selection tools intentionally live on the floating selection toolbar. The docked
 * left rail is reserved for shared brush settings, and selection results are shown
 * by the Tile Inspector rather than duplicated in the bottom context drawer.
 */
class SingleSelectToolPlugin : StudioToolPlugin {
    override fun id(): String = ID

    override fun name(): String = "Single Select"

    override fun description(): String =
        "Click a tile to select exactly one tile for inspection."

    override fun icon(): String = StudioIcons.SELECT

    override fun toolId(): String = ENGINE_TOOL_ID

    override fun shortcut(): String = "S"

    override fun railPriority(): Int = 10

    override fun category(): String = "Selection"

    override fun surfaces(): MutableSet<StudioToolPlugin.ToolSurface> =
        EnumSet.of(StudioToolPlugin.ToolSurface.FLOATING_TOOLBAR)

    override fun hasContextDrawerContent(): Boolean = false

    override fun renderContextDrawer(context: StudioPanelContext?) {
        // Intentionally empty: the Tile Inspector owns selection-result presentation.
    }

    companion object {
        const val ID = "studio.tool.select.single"
        const val ENGINE_TOOL_ID = "selection.single"
    }
}
